src/ledger/database/interface.zig

```diff
@@ -536,5 +540,178 @@ pub fn testDatabase(comptime Impl: fn ([]const ColumnFamily) type) type {
+            var db = try DB.open(allocator, logger, path);
+            defer db.deinit();
+
+            try db.put(cf1, 40, .{ .hello = 44 });
+            try db.put(cf1, 10, .{ .hello = 111 });
+            try db.put(cf1, 30, .{ .hello = 33 });
+            try db.put(cf1, 20, .{ .hello = 222 });
+
+            var iter = try db.iterator(cf1, .reverse, 5);
+            defer iter.deinit();
+
+            try std.testing.expectEqual(null, try iter.next());
+        }
+
+        test "iterator forward empty" {
+            const allocator = std.testing.allocator;
+            const path = test_dir ++ @src().fn_name;
+            try ledger.tests.freshDir(path);
+            var db = try DB.open(allocator, logger, path);
+            defer db.deinit();
+
+            var iter = try db.iterator(cf1, .forward, 1);
+            defer iter.deinit();
+
+            try std.testing.expectEqual(null, try iter.next());
+        }
+
+        test "iterator reverse empty" {
+            const allocator = std.testing.allocator;
+            const path = test_dir ++ @src().fn_name;
+            try ledger.tests.freshDir(path);
+            var db = try DB.open(allocator, logger, path);
+            defer db.deinit();
+
+            var iter = try db.iterator(cf1, .reverse, 1);
+            defer iter.deinit();
+
+            try std.testing.expectEqual(null, try iter.next());
+        }
+
+        test "iterator forward empty with null start" {
+            const allocator = std.testing.allocator;
+            const path = test_dir ++ @src().fn_name;
+            try ledger.tests.freshDir(path);
+            var db = try DB.open(allocator, logger, path);
+            defer db.deinit();
+
+            var iter = try db.iterator(cf1, .forward, null);
+            defer iter.deinit();
+
+            try std.testing.expectEqual(null, try iter.next());
+        }
+
+        test "iterator reverse empty with null start" {
+            const allocator = std.testing.allocator;
+            const path = test_dir ++ @src().fn_name;
+            try ledger.tests.freshDir(path);
+            var db = try DB.open(allocator, logger, path);
+            defer db.deinit();
+
+            var iter = try db.iterator(cf1, .reverse, null);
+            defer iter.deinit();
+
+            try std.testing.expectEqual(null, try iter.next());
+        }
+
+        test "WriteBatch.deleteRange" {
+            const allocator = std.testing.allocator;
+            const path = test_dir ++ @src().fn_name;
+            try ledger.tests.freshDir(path);
+            var db = try DB.open(allocator, logger, path);
+            defer db.deinit();
+
+            try db.put(cf1, 40, .{ .hello = 44 });
+            try db.put(cf1, 10, .{ .hello = 111 });
+            try db.put(cf1, 30, .{ .hello = 33 });
+            try db.put(cf1, 20, .{ .hello = 222 });
+
+            var batch = try db.initWriteBatch();
+            defer batch.deinit();
+            try batch.deleteRange(cf1, 0, 100);
```

Comment:
Nit: if there is a bug in `deleteRange` where it deletes everything instead of what's in between start and end index, this test won't catch this. Having `try batch.deleteRange(cf1, 20, 35);` might be better to confirm 20 and 30 were deleted but 10 and 40 were not